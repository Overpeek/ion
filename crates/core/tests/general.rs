use std::process::exit;

use ion_core::Ion;

#[test]
fn invalid_token() {
    let code = r#"x = \"#;
    let ion = Ion::new();
    let err = ion.parse_str(code).unwrap_err();

    insta::assert_display_snapshot!(err.pretty_print(false, code, "<code>"));
}

#[test]
fn unexpected_eof() {
    let code = r#"x ="#;
    let ion = Ion::new();
    let err = ion.parse_str(code).unwrap_err();

    insta::assert_display_snapshot!(err.pretty_print(false, code, "<code>"));
}

#[test]
fn unexpected_token_1() {
    let code = r#"x = ="#;
    let ion = Ion::new();
    let err = ion.parse_str(code).unwrap_err();

    insta::assert_display_snapshot!(err.pretty_print(false, code, "<code>"));
}

#[test]
fn unexpected_token_2() {
    let code = r#"""#;
    let ion = Ion::new();
    let err = ion.parse_str(code).unwrap_err();

    insta::assert_display_snapshot!(err.pretty_print(false, code, "<code>"));
}

#[test]
fn correct_code() {
    let code = r#"
        x = 4;
    fn y() { x = 2; };
    y();"#;

    let ion = Ion::new();
    ion.parse_str(code).unwrap_or_else(|err| {
        eprintln!("{}", err.pretty_print(true, code, "<code>"));
        exit(-1);
    });
}
