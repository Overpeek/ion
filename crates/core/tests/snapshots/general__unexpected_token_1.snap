---
source: crates/core/tests/general.rs
expression: "err.pretty_print(false, code, \"<code>\")"
---
error: Unexpected token, expected one of "(", "false", "fn(", "true", r#"\"[^\"]*\""#, r#"[a-zA-Z_][\\w_]*"#, r#"\\.\\d[\\d_]*"#, r#"\\d[\\d_]*"#, r#"\\d[\\d_]*\\."# or r#"\\d[\\d_]*\\.\\d[\\d_]*"# --> <code>:1:5
  |
1 | x = =
  |     ^

